packages/framework/src/types/validator.types.ts

```diff
@@ -19,15 +21,15 @@ export type ValidateResult<T> =
       data: T;
     };
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export interface Validator<T_Schema extends Schema = Schema> {
+export type Validator<T_Schema extends Schema = Schema> = {
   validate: <
     T_Unvalidated extends Record<string, unknown> = FromSchemaUnvalidated<T_Schema>,
     T_Validated extends Record<string, unknown> = FromSchema<T_Schema>,
   >(
     data: T_Unvalidated,
     schema: T_Schema
   ) => Promise<ValidateResult<T_Validated>>;
-  canHandle: (schema: Schema) => schema is T_Schema;
+  canHandle: (schema: Schema) => Promise<boolean>;
   transformToJsonSchema: (schema: T_Schema) => Promise<JsonSchema>;
-}
+  requiredImports: readonly ImportRequirement[];
```

Comment:
Each validator now declares a list of required imports that can be checked concretely at runtime.

I've found this to be the most reliable approach as different build + runtiem environments (CJS/ESM + `next dev`/`next dev --turbo` can handle import results differently - sometimes they throw during `await import(...)`, other times they resolve with empty exports. We handle both failure cases with `checkDependencies` now and we can dynamically construct the list of missing imports, providing an excellent `npm install x` DX.